contracts/helpers/TransferHelper.sol

```diff
@@ -0,0 +1,166 @@
+            conduitController
+        );
+        (_CONDUIT_CREATION_CODE_HASH, ) = controller.getConduitCodeHashes();
+
+        // Set the supplied conduit controller as an immutable.
+        _CONDUIT_CONTROLLER = controller;
+    }
+
+    /**
+     * @notice Transfer multiple items to a single recipient.
+     *
+     * @param items      The items to transfer.
+     * @param recipient  The address the items should be transferred to.
+     * @param conduitKey The key of the conduit through which the bulk transfer
+     *                   should occur.
+     *
+     * @return magicValue A value indicating that the transfers were successful.
+     */
+    function bulkTransfer(
+        TransferHelperItem[] calldata items,
+        address recipient,
+        bytes32 conduitKey
+    ) external override returns (bytes4 magicValue) {
+        // Retrieve total number of transfers and place on stack.
+        uint256 totalTransfers = items.length;
+
+        // If no conduitKey is given, use TokenTransferrer to perform transfers.
+        if (conduitKey == bytes32(0)) {
+            // Skip overflow checks: all for loops are indexed starting at zero.
+            unchecked {
+                // Iterate over each transfer.
+                for (uint256 i = 0; i < totalTransfers; ++i) {
+                    // Retrieve the transfer in question.
+                    TransferHelperItem calldata item = items[i];
+
+                    // Perform a transfer based on the transfer's item type.
+                    // Revert if item being transferred is not a native token.
+                    if (item.itemType == ConduitItemType.NATIVE) {
+                        revert InvalidItemType();
+                    } else if (item.itemType == ConduitItemType.ERC20) {
+                        _performERC20Transfer(
+                            item.token,
+                            msg.sender,
+                            recipient,
+                            item.amount
+                        );
+                    } else if (item.itemType == ConduitItemType.ERC721) {
+                        _performERC721Transfer(
+                            item.token,
+                            msg.sender,
+                            recipient,
+                            item.identifier
+                        );
+                    } else {
+                        _performERC1155Transfer(
+                            item.token,
+                            msg.sender,
+                            recipient,
+                            item.identifier,
+                            item.amount
+                        );
+                    }
+                }
+            }
+        }
+        // Otherwise, a conduitKey was provided.
+        else {
+            // Derive the conduit address from the deployer, conduit key
+            // and creation code hash.
+            address conduit = address(
```

Comment:
I wonder if it would make sense instead having a public function for this on the conduit, though that has a call cost.